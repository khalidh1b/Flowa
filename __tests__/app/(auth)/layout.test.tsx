import React from "react";
import { render } from "@testing-library/react";
import AuthLayout from "@/app/(auth)/layout";

jest.mock("@clerk/nextjs", () => ({
  ClerkLoading: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="clerk-loading">{children}</div>
  ),
  ClerkLoaded: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="clerk-loaded">{children}</div>
  ),
  SignedIn: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="signed-in">{children}</div>
  ),
  SignedOut: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="signed-out">{children}</div>
  ),
}));


describe("AuthLayout", () => {
  it("renders correctly (snapshot) for loading state", () => {
    const { asFragment, getByTestId } = render(<AuthLayout>Test Its Content</AuthLayout>);
    expect(getByTestId("clerk-loading")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly (snapshot) for signed in state", () => {
    const { asFragment, getByTestId } = render(<AuthLayout>Test Its Content</AuthLayout>);
    expect(getByTestId("signed-in")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly (snapshot) for signed out state", () => {
    const { asFragment, getByTestId } = render(<AuthLayout>Test Its Content</AuthLayout>);
    expect(getByTestId("signed-out")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});