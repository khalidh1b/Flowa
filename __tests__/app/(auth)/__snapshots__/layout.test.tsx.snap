// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`AuthLayout renders correctly (snapshot) for loading state 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center pt-40"
  >
    <div
      data-testid="clerk-loading"
    >
      <div>
        <svg
          class="lucide lucide-loader-circle animate-spin w-7 w-7"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 12a9 9 0 1 1-6.219-8.56"
          />
        </svg>
         Loading...
      </div>
    </div>
    <div
      data-testid="clerk-loaded"
    >
      <div
        data-testid="signed-in"
      >
        <div>
          Test Its Content
        </div>
      </div>
      <div
        data-testid="signed-out"
      >
        <div>
          Test Its Content
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AuthLayout renders correctly (snapshot) for signed in state 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center pt-40"
  >
    <div
      data-testid="clerk-loading"
    >
      <div>
        <svg
          class="lucide lucide-loader-circle animate-spin w-7 w-7"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 12a9 9 0 1 1-6.219-8.56"
          />
        </svg>
         Loading...
      </div>
    </div>
    <div
      data-testid="clerk-loaded"
    >
      <div
        data-testid="signed-in"
      >
        <div>
          Test Its Content
        </div>
      </div>
      <div
        data-testid="signed-out"
      >
        <div>
          Test Its Content
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AuthLayout renders correctly (snapshot) for signed out state 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center pt-40"
  >
    <div
      data-testid="clerk-loading"
    >
      <div>
        <svg
          class="lucide lucide-loader-circle animate-spin w-7 w-7"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 12a9 9 0 1 1-6.219-8.56"
          />
        </svg>
         Loading...
      </div>
    </div>
    <div
      data-testid="clerk-loaded"
    >
      <div
        data-testid="signed-in"
      >
        <div>
          Test Its Content
        </div>
      </div>
      <div
        data-testid="signed-out"
      >
        <div>
          Test Its Content
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
